cmd/restic/cmd_dump.go

```diff
@@ -188,7 +173,8 @@ func runDump(opts DumpOptions, gopts GlobalOptions, args []string) error {
 		Exitf(2, "loading tree for snapshot %q failed: %v", snapshotIDString, err)
 	}
 
-	err = printFromTree(ctx, tree, repo, "/", splittedPath, wd)
+	d := dump.New(opts.Archive, repo, os.Stdout)
```

Comment:
What about letting `dump.New` handle the check which output formats are supported? That way the check would only be in a single place.

Reply:
We need the repository to be open for dump.New and I want to validate the arguments before potentially showing a password prompt.